Sources/MongoSwift/MongoConnectionString.swift

```diff
@@ -119,6 +119,31 @@ public struct MongoConnectionString: Codable, LosslessStringConvertible {
         }
     }
 
+    private static func percentDecode(from: String?) throws -> String? {
+        if from == nil {
```

Comment:
style nit: I think using `guard let` to unwrap `from` would be more idiomatic here:
```swift
guard let from = from else {
   return nil
}
// use non-optional `from` down here...
```

generally, when you are checking some precondition for a parameter that is necessary for the rest of the method to execute, `guard` is appropriate